### PR TITLE
aws: Update to version 2.0.23

### PR DESCRIPTION
**Pull request by xfournet**
AWS CLI v2 is now official

### CODE REVIEW

**Comment by Ash258** [comment]
Why there is no changelog?

image

**Comment by xfournet** [comment]
I cannot say how they handle that on github. The tag is here https://github.com/aws/aws-cli/releases/tag/2.0.0 and the announcement is here https://aws.amazon.com/blogs/developer/aws-cli-v2-is-now-generally-available/

**Comment by xfournet** [comment]
Ah found it in the v2 branch: https://github.com/aws/aws-cli/blob/v2/CHANGELOG.rst

**Comment by Ash258** [comment]
@rasa Wait. There has to be everything fixed.

- API in checkver
- backslashes
- bin is faulty

I will do proper review when I have time or request changest for it on your own

**Comment by xfournet** [comment]
@Ash258 please let me know what changes i should make.
* API in checkver : it's due that v1.x tags have been created after v2.x tag. The github checkver return the last tag based on it's date, which doesn't work in this case. The github API permit to retrieve last tag from a semantic perspective (so we have the 2.0.0 version returned as expected). Please advise if there is a better approach to this problem ?
* backslashes : i don't have backslashes in my diff, could you explain ?
* bin is faulty : i don't see what is the problem, it's working here. What i have missed ?

**Comment by xfournet** [comment]
Removed the unnecessary [ ] for bin

File: bucket/aws.json [hunk 1]
```diff
@@ -1,22 +1,17 @@
...
-    "bin": [
-        [
-            "runtime/python.exe",
-            "aws",
-            "-m awscli"
-        ]
-    ],
+    "url": "https://awscli.amazonaws.com/AWSCLIV2-2.0.0.msi",
+    "hash": "47602fe3e6d8348609fa01ff535cd6c3708d3f8c942e90e6cf3027241fb8cbca",
+    "extract_dir": "Amazon/AWSCLIV2",
```
Comment (Ash258) [inline bucket/aws.json]:
```suggestion
    "extract_dir": "Amazon\\AWSCLIV2",
```

File: bucket/aws.json [hunk 2]
```diff
@@ -1,22 +1,17 @@
...
+    "bin": "aws.exe",
     "checkver": {
-        "github": "https://github.com/aws/aws-cli"
+        "url": "https://api.github.com/repos/aws/aws-cli/tags",
+        "jsonpath": "$[0].name"
     },
     "autoupdate": {
-        "url": "https://s3.amazonaws.com/aws-cli/AWSCLI64PY3-$version.msi"
+        "url": "https://awscli.amazonaws.com/AWSCLIV2-$version.msi"
```
Comment (Ash258) [inline bucket/aws.json]:
```suggestion
        "architecture": {
            "64bit": {
                "url": "https://awscli.amazonaws.com/AWSCLIV2-$version.msi"
            }
        }
```

File: bucket/aws.json [hunk 3]
```diff
@@ -1,22 +1,17 @@
...
-    "extract_dir": "Amazon/AWSCLI",
-    "bin": [
-        [
-            "runtime/python.exe",
-            "aws",
-            "-m awscli"
-        ]
-    ],
+    "url": "https://awscli.amazonaws.com/AWSCLIV2-2.0.0.msi",
+    "hash": "47602fe3e6d8348609fa01ff535cd6c3708d3f8c942e90e6cf3027241fb8cbca",
```
Comment (Ash258) [inline bucket/aws.json]:
```suggestion
    "architecture": {
        "64bit": {
            "url": "https://awscli.amazonaws.com/AWSCLIV2-2.0.0.msi",
            "hash": "47602fe3e6d8348609fa01ff535cd6c3708d3f8c942e90e6cf3027241fb8cbca"
        }
    },
```

**Comment by xfournet** [comment]
Required changed have been done.
My motivation to upgrade is to be able to use AWS SSO which is a great security improvement for us. I will monitor new releases to adapt checkver accordingly when required.

**Comment by Livven** [comment]
@Ash258
> * I do not see a reason to update unless it is properly communicate and [aws/aws-cli#4961](https://github.com/aws/aws-cli/issues/4961) is resolved.

The official communication is the [already linked blog post](https://aws.amazon.com/blogs/developer/aws-cli-v2-is-now-generally-available/). The download channel for the vast majority of users will be the [official website](https://aws.amazon.com/cli/), which provides links to the v2 installers. It doesn't get more officially communicated than that.

It does seem weird that v2 has not been merged into the master branch in the Github repo, but it's probably safe to assume that they now see the official website as the proper entrypoint for most people so the Github repo is not as important anymore.

Would be great to get this merged :)

**Comment by xfournet** [comment]
@rasa @Ash258 please could you let me know what need to be done to get it merged ?
aws CLI v2 is now officially released since 4 months and scoop is still delivering the v1 (btw even the v1 is not up to date : 1.18.44 in Scoop while the latest is 1.18.78)
the PR is ready, autoupdate using github API has worked everytime i used it, so i don't see any blocker here

**Comment by rasa** [comment]
Superceded by #1395?

**Pull request closed without merging**